pkg/kubelet/kubelet_pods.go

```diff
@@ -1035,20 +1035,14 @@ func (kl *Kubelet) HandlePodCleanups() error {
 
 	kl.removeOrphanedPodStatuses(allPods, mirrorPods)
 	// Note that we just killed the unwanted pods. This may not have reflected
-	// in the cache. We need to bypass the cache to get the latest set of
-	// running pods to clean up the volumes.
-	// TODO: Evaluate the performance impact of bypassing the runtime cache.
-	runningPods, err = kl.containerRuntime.GetPods(false)
-	if err != nil {
-		klog.Errorf("Error listing containers: %#v", err)
-		return err
-	}
+	// in the cache. Getting the latest set of running pods to clean up the volumes, considering the
+	// intersection with desired Pods.
```

Comment:
Sorry for the late response. I reviewed it a while back, but forgot to send the review response. 

I am not sure I understand the issue you addressed here. Based on the pr, looks like with this pr, we should bypass the runtime cache for cleanup, but later in method cleanupOrphanedPodDirs(...), we still loop through runningPods. In addition to that, the cleanup logic below create the union of runningPods and desirePods, instead of intersection.

Reply:
The loop over runningPods is to compute the proper allPods Set.
I have modified the comment prior to calling cleanupOrphanedPodDirs so that it is consistent with what the func does.

Reply:
@dchen1107 
The savings brought by this PR are in the reduction of runtimeService.ListPodSandbox call.